runners/direct-java/src/main/java/org/apache/beam/runners/direct/TestStreamEvaluatorFactory.java

```diff
@@ -53,8 +54,22 @@
  * The {@link TransformEvaluatorFactory} for the {@link TestStream} primitive.
  */
 class TestStreamEvaluatorFactory implements TransformEvaluatorFactory {
-  private final AtomicBoolean inUse = new AtomicBoolean(false);
-  private final AtomicReference<Evaluator<?>> evaluator = new AtomicReference<>();
+  /**
+   * A map from {@link TestStream} application to an {@link Optional} {@link Evaluator} for that
+   * application. At most one evaluator is created for an AppliedPTransform and it is used by at
+   * most one thread at a time.
+   *
+   * <p>For each {@link AppliedPTransform} in this map:
+   *
+   * <ul>
+   * <li>If there is no associated value, then no evaluator has been created yet.
+   * <li>If the value is {@code Optional.absent()} then the evaluator is currently in use.
+   * <li>If the value is {@code Optional.present()} then the contained evaluator is available for
+   *     use.
+   * </ul>
+   */
```

Comment:
This Javadoc can probably go away/move to the SingleInstanceLimitedConcurrenceCache and/or be simplified.